block/rollkit-minimal-header.md

```diff
@@ -0,0 +1,192 @@
+# Rollkit Minimal Header
+
+## Abstract
+
+This document specifies a minimal header format for Rollkit, designed to eliminate the dependency on CometBFT's header format. This new format can then be used to produce an execution layer tailored header if needed. For example, the new ABCI Execution layer can have an ABCI-specific header for IBC compatibility. This allows Rollkit to define its own header structure while maintaining backward compatibility where necessary.
+
+## Protocol/Component Description
+
+The Rollkit minimal header is a streamlined version of the traditional header, focusing on essential information required for block processing and state management for rollup nodes. This header format is designed to be lightweight and efficient, facilitating faster processing and reduced overhead.
+
+### Rollkit Minimal Header Structure
+
+```
+┌─────────────────────────────────────────────┐
+│             Rollkit Minimal Header          │
+├─────────────────────┬───────────────────────┤
+│ ParentHash          │ Hash of previous block│
+├─────────────────────┼───────────────────────┤
+│ Height              │ Block number          │
+├─────────────────────┼───────────────────────┤
+│ Timestamp           │ Creation time         │
+├─────────────────────┼───────────────────────┤
+│ ChainID             │ Chain identifier      │
+├─────────────────────┼───────────────────────┤
+│ DataCommitment      │ Pointer to block data │
+│                     │ on DA layer           │
+├─────────────────────┼───────────────────────┤
+│ StateRoot           │ State commitment      │
+├─────────────────────┼───────────────────────┤
+│ ExtraData           │ Additional metadata   │
+│                     │ (e.g. sequencer info) │
+└─────────────────────┴───────────────────────┘
+```
+
+## Message Structure/Communication Format
+
+The header is defined in GoLang as follows:
+
+```go
+// Header struct focusing on header information
+type Header struct {
+    // Hash of the previous rollup block header.
+    ParentHash Hash
+    // Height represents the block height (aka block number) of a given header
+    Height uint64
+    // Block creation timestamp
+    Timestamp uint64
+    // The Chain ID
+    ChainID string
+    // Pointer to location of associated block data aka transactions in the DA layer
+    DataCommitment Hash
```

Comment:
can we make this `[][]byte`. because at least for the celestia DA, we plan on putting the []da.ID where da.ID is []byte and it is constructed using da height + da commitment.

Reply:
or simply `[]byte`